run-test.sh

```diff
@@ -253,8 +250,6 @@ coreclr_code_coverage()
 # Parse arguments
 
 ((serverGC = 0))
-OuterLoop="-notrait category=outerloop"
-IgnoreForCI=""
```

Comment:
> Where do RunTests.sh/RunTests.cmd come from? It looks like they're generated by some build targets (dotnet/buildtools#649 ?) but it isn't clear to me what exactly their purpose is.

Correct, they are generated by targets in BuildTools.

Their purpose is two fold:

1. Actually finish setting up the test drop (it does this by linking files from the packages folder into the test execution root.  When you build with /p:SkipTests=true, you no longer have libcoreclr.so and friends in the test folder, instead you just have a shell script that will link them in later).
2. Invoke xunit with the correct arguments.

The reason this script exists is because we want to Unify test execution between Helix and Jenkins.  This will also allow different types of builds (running against 4.6.2, running against .NET Native) to produce whatever RunTests.sh/RunTests.cmd they want and automation which runs the test does not need to update).

> Also, does this remove our ability to pass Outerloop and IgnoreForCI explicitly from the netci.groovy for running tests? Because they're removed from run-test.sh here but they're still being passed in as input in the netci file (line 448).

You can pass them, but they are ignored.  I plan to clean this up, but due to how netci.groovy changes only updated CI jobs after they are merged, I have to keep us in this intermediate state.  The flags are now part of RunTests.cmd/sh which is generated at build time (so if you pass /p:Outerloop=true to your build, the generated file will not exclude the outerloop category).